src/utils/DependencyUtils.ts

```diff
@@ -11,3 +21,49 @@ export const splitToNameAndVersion = (dependencyString: string): [string, string
     const fullName = parts.join('-');
     return [fullName, version];
 };
+
+export async function getFullDependencyList(combos: ThunderstoreCombo[]) {
```

Comment:
IDK how much effort it would require to get all necessary things mocked, but having unit tests for this method would be great. It should cover at least the use cases of installing a mod, installing a mod pack, updating all for profile containing both mods and modpacks, and importing a profile containing mods and mod packs.

The test coverage overall is poor, but in a refactor like this, having those tests in place would be great.

Reply:
I'll look into this, I'll create another PR with the tests and chain it to this one.